### PR TITLE
Update C# executable template to support c# executable

### PR DESCRIPTION
**Pull request by YingChen1996**
Update C# executable template to support c# executable.

This pull request introduces changes to the `promptflow` package, specifically to the way the package handles executable files and their dependencies. The changes primarily involve updates to the file structure and modifications to several Python scripts.

File structure updates:

* [`src/promptflow/promptflow/_sdk/data/executable/README.md`](diffhunk://#diff-e78e25d4a7e068f2b2b0f73b6bfebabe1a2a66435156bc19721270794c027fa8R7-R21): Added instructions on how to add flow dependencies in the `requirements.txt` file before building the executable.

Python script modifications:

* [`src/promptflow/promptflow/_sdk/data/executable/app.py`](diffhunk://#diff-8eea9c3a8725c905cf1ebfce3b07e7093715d311681d362eaec9e970a681996fL4-R17): The import order was changed, and the `sys.argv` list was reformatted for readability. [[1]](diffhunk://#diff-8eea9c3a8725c905cf1ebfce3b07e7093715d311681d362eaec9e970a681996fL4-R17) [[2]](diffhunk://#diff-8eea9c3a8725c905cf1ebfce3b07e7093715d311681d362eaec9e970a681996fL32-R41)
* [`src/promptflow/promptflow/_sdk/data/executable/app.spec.jinja2`](diffhunk://#diff-0a82f38f8afa4f1e9640adf03828adb86deeef57043ed3b3f785fd952c636f52R2-R8): The recursion limit was increased, and the `datas` list was updated to include `config.json`.
* [`src/promptflow/promptflow/_sdk/data/executable/main.py.jinja2`](diffhunk://#diff-6d16dfee69c366229fa2e1d5a1e6c81842b462f0377a248dd9614c507ca3f5b3L1-L142): The entire file was removed.
* [`src/promptflow/promptflow/_sdk/operations/_flow_operations.py`](diffhunk://#diff-afdd40a5d0519512dcf9be48bd46c4caaa2291b808687de77896989af63f47e4L526-L543): The `flow_inputs_params` variable was removed, and a `config.json` file was created to store various configuration details.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_cli/_pf/_init_entry_generators.py [hunk 1]
```diff
@@ -238,10 +238,10 @@ def flow_inputs(self):
         if self.is_chat_flow:
             results = {}
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_cli/_pf/_init_entry_generators.py:239]: please add more PR description

Comment (YingChen1996) [reply src/promptflow/promptflow/_cli/_pf/_init_entry_generators.py]: added

File: src/promptflow/promptflow/_sdk/operations/_flow_operations.py [hunk 2]
```diff
@@ -523,24 +523,37 @@ def _build_as_executable(
...
+        chat_output_name = next(
+            filter(
+                lambda key: executable.outputs[key].is_chat_output,
+                executable.outputs.keys(),
+            ),
+            None,
+        )
         label = "Chat" if is_chat_flow else "Run"
+        is_streaming = True if is_chat_flow else False
+        config_content = {
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/operations/_flow_operations.py:537]: what's the core change that's related to C# support?

Comment (YingChen1996) [reply src/promptflow/promptflow/_sdk/operations/_flow_operations.py:537]: use main.py as executable template, remove main.py.jinja

**Review by wangchao1230 — APPROVED** [review]

**Review by crazygao — APPROVED** [review]

**Pull request merged**